File: LV1/1.5/onechicken/onechicken.rb
a, b = gets.split.map(&:to_i)
diff = a - b

p = 'piece'
p += 's' if diff.abs > 1

if diff > 0
  puts "Dr. Chaz needs #{diff} more #{p} of chicken!"
else
  puts "Dr. Chaz will have #{-diff} #{p} of chicken left over!"
end

File: LV1/1.5/onechicken/onechicken.rs
fn main() {
    let stdin = std::io::stdin();
    let mut buff = String::new();
    stdin.read_line(&mut buff).unwrap();

    let num: Vec<i32> = buff
        .split_whitespace()
        .map(|n| n.parse().unwrap())
        .collect();

    let diff = num[0] - num[1];
    let p = if diff.abs() != 1 { "pieces" } else { "piece" };

    if diff > 0 {
        println!("Dr. Chaz needs {} more {} of chicken!", diff, p);
    } else {
        println!("Dr. Chaz will have {} {} of chicken left over!", -diff, p);
    }
}
